dot/state/block.go

```diff
@@ -291,35 +247,73 @@ func (bs *BlockState) SetHeader(header *types.Header) error {
 	return nil
 }
 
-// SetBlock will add a block to the DB
-func (bs *BlockState) SetBlock(block *types.Block) error {
-	// Add the blockHeader to the DB
-	err := bs.SetHeader(block.Header)
+// GetBlockBody will return Body for a given hash
+func (bs *BlockState) GetBlockBody(hash common.Hash) (*optional.Body, error) {
+	data, err := bs.db.Get(blockBodyKey(hash))
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	blockData := &types.BlockData{
-		Hash:   block.Header.Hash(),
-		Header: block.Header.AsOptional(),
-		Body:   block.Body.AsOptional(),
-	}
-	return bs.SetBlockData(blockData)
+	return &optional.Body{Exists: true, Value: data}, nil
 }
 
-// SetBlockData will set the block data using given hash and blockData into DB
-func (bs *BlockState) SetBlockData(blockData *types.BlockData) error {
+// SetBlockBody will add a block body to the db
+func (bs *BlockState) SetBlockBody(hash common.Hash, body *optional.Body) error {
```

Comment:
same here, there is no point in putting an empty optional in the DB. argument should be `types.Body`

Reply:
updated.